checker/src/test/java/org/checkerframework/checker/test/junit/ResourceLeakExtraIgnoredExceptionsTest.java

```diff
@@ -0,0 +1,25 @@
+package org.checkerframework.checker.test.junit;
+
+import java.io.File;
+import java.util.List;
+import org.checkerframework.checker.resourceleak.ResourceLeakChecker;
+import org.checkerframework.framework.test.CheckerFrameworkPerDirectoryTest;
+import org.junit.runners.Parameterized;
+
+public class ResourceLeakExtraIgnoredExceptionsTest extends CheckerFrameworkPerDirectoryTest {
+  public ResourceLeakExtraIgnoredExceptionsTest(List<File> testFiles) {
+    super(
+        testFiles,
+        ResourceLeakChecker.class,
+        "resourceleak-extraignoredexceptions",
+        "-AresourceLeakIgnoredExceptions=default,java.lang.IllegalStateException",
```

Comment:
I guess our new syntax could be a bit awkward if your first ignored exception type is exact, since then you'll have to write `-AresourceLeakIgnoredExceptions==exception.Type`; not sure if this is a significant concern.

Thinking more, I think I was confused before.  I think we only supported exact type matching before, and this PR adds support for matching all exception subtypes.  Is that correct @Calvin-L?  Maybe I was wrong about what would be a more common scenario...

Reply:
>  I think we only supported exact type matching before, and this PR adds support for matching all exception subtypes.

That's right.

> Maybe I was wrong about what would be a more common scenario...

I actually think your first thought was right. I suspect that matching all subtypes will be the more common scenario, if only because it is easier to understand. Matching on exact type has very surprising semantics.

I haven't checked this recently, but if memory serves---

With no special options, the RLC allows this code:

```java
r = alloc();
method();
r.close();
```

but does _not_ allow this code:

```java
r = alloc();
try {
  method();
} catch (IllegalStateException e) {
  throw e;
}
r.close();
```

The different handling is extremely confusing, because those snippets have essentially identical run-time behavior. The call to `method()` always had the option to throw `IllegalStateException`, regardless of whether we caught and re-threw it. Since `IllegalStateException extends RuntimeException` and `RuntimeException` is ignored by default, shouldn't the `catch` block in the second snippet be treated as unreachable?

For that reason, I think that matching all subtypes is more intuitive. With `-AresourceLeakIgnoredExceptions=java.lang.Error,java.lang.RuntimeException` the two snippets are both accepted.

Reply:
Hrm...interesting.  By your argument, I think we should then also change how the default list of exceptions is handled (not based on exact name, but on any subtype).

But!  When adding the default ignored exception functionality, in my mind I was primarily thinking about cases where the exceptional edge for the call went directly to the method exit.  For your second example, I'm honestly not sure if we should be ignoring the potential exceptions from the call to `method()` or not.  Somehow, the presence of a `catch` block to me implies that the developer cares more about the possibility of that exception being thrown; so maybe we should not ignore the possibility?

But in any case, this point is somewhat independent of this PR.  Even now, writing this code, I think the exceptional possibility will be ignored by the RLC:
```java
r = alloc();
try {
  method();
} catch (RuntimeException e) {
  throw e;
}
r.close();
```

So I think the question for this PR is whether the default list of ignored exceptions should also match all subtypes, and my instinct is to say yes.  @kelloggm any thoughts on this?

Reply:
> I think the question for this PR is whether the default list of ignored exceptions should also match all subtypes, and my instinct is to say yes

Yes, I agree: the default list of ignored exceptions should match subtypes. That's a change to the RLC's externally-facing behavior, however, so this PR also needs a `CHANGELOG` entry.

> Even now, writing this code, I think the exceptional possibility will be ignored by the RLC

I agree, I think this will be ignored. I agree with your intuition that this is *weird*: it would be surprising to me, as a user, that the RLC ignores the possibility of a `catch` block that I explicitly add. A separate PR to address this is a good idea (and I see you opened a separate issue, too).

Reply:
> the default list of ignored exceptions should match subtypes

Do you want me to add that change to this PR? (In theory it's an orthogonal change, and I would opt to do it in a follow-up PR if given the option.)

Reply:
A follow-up PR is fine, good point.  Thanks!